packages/selection/frontend/sirius-components-selection/src/SelectionDialogTreeView.tsx

```diff
@@ -39,16 +45,71 @@ export const SelectionDialogTreeView = ({
   editingContextId,
   treeDescriptionId,
   variables,
-  enableMultiSelection,
+  onTreeItemClick,
+  selectedTreeItemIds,
 }: SelectionDialogTreeViewProps) => {
   const { classes } = useTreeStyle();
   const [state, setState] = useState<SelectionDialogTreeViewState>(initialState);
 
   const treeId = `selection://?treeDescriptionId=${encodeURIComponent(treeDescriptionId)}${encodeVariables(variables)}`;
   const { tree } = useSelectionDialogTreeSubscription(editingContextId, treeId, state.expanded, state.maxDepth);
 
-  const onExpandedElementChange = (expanded: string[], maxDepth: number) => {
-    setState((prevState) => ({ ...prevState, expanded, maxDepth }));
+  const {
+    getExpandAllTreePath,
+    loading: expandAllTreePathLoading,
+    data: expandAllTreePathData,
+  } = useExpandAllTreePath();
+
+  useEffect(() => {
+    if (!expandAllTreePathLoading) {
```

Comment:
We can remove at least two "if" from this function without a change in behavior, it would improve the readability

Reply:
Done